File: hh-back/src/main/java/gloit/hiperionida/helios/model/CamionModel.java
package gloit.hiperionida.helios.model;


import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "camion")
public class CamionModel extends AbsVehiculoModel {
    private String numeroChasis;
    private String numeroMotor;

    public CamionModel() {
    }

    public CamionModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, Integer anio, Integer cantidadNeumaticos, String marcaModelo, String notas, String patente, String peso, String numeroChasis, String numeroMotor) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada, anio, cantidadNeumaticos, marcaModelo, notas, patente, peso);
        this.numeroChasis = numeroChasis;
        this.numeroMotor = numeroMotor;
    }
}

File: hh-front/src/models/camion_model.js
import { UsuarioModel } from './usuario_model'
import { SeguroModel } from './seguro_model'

export class CamionModel {
  constructor () {
    this.cantidadNeumaticos = ''
    this.marca = ''
    this.modelo = ''
    this.anio = ''
    this.patente = ''
    this.peso = ''
    this.numeroChasis = ''
    this.numeroMotor = ''
    this.pesoArrastre = ''
    this.seguro = new SeguroModel()
    this.servicios = new Array([])
    this.neumaticos = new Array([])

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}
